pkg/alertmanager/multitenant_test.go

```
@@ -601,6 +608,97 @@ func TestMultitenantAlertmanager_deleteUnusedLocalUserState(t *testing.T) {
+		cfg.Persister.Interval = 500 * time.Millisecond
+
+		am, err := createMultitenantAlertmanager(cfg, nil, nil, alertStore, ringStore, nil, log.NewLogfmtLogger(os.Stdout), reg)
+		require.NoError(t, err)
+		t.Cleanup(func() {
+			require.NoError(t, services.StopAndAwaitTerminated(ctx, am))
+		})
+		require.NoError(t, services.StartAndAwaitRunning(ctx, am))
+
+		return am
+	}
+
+	am1ZoneA := createInstance(1, "zoneA")
+	am2ZoneA := createInstance(2, "zoneA")
+	am1ZoneB := createInstance(3, "zoneB")
+
+	// Configure the users and wait for the state persister to write some state for both.
+	{
+		require.NoError(t, alertStore.SetAlertConfig(ctx, alertspb.AlertConfigDesc{
+			User:      user1,
+			RawConfig: simpleConfigOne,
+			Templates: []*alertspb.TemplateDesc{},
+		}))
+		require.NoError(t, alertStore.SetAlertConfig(ctx, alertspb.AlertConfigDesc{
+			User:      user2,
+			RawConfig: simpleConfigOne,
+			Templates: []*alertspb.TemplateDesc{},
+		}))
+		require.NoError(t, alertStore.SetAlertConfig(ctx, alertspb.AlertConfigDesc{
+			User:      user3,
+			RawConfig: simpleConfigOne,
+			Templates: []*alertspb.TemplateDesc{},
+		}))
+
+		err := am1ZoneA.loadAndSyncConfigs(context.Background(), reasonPeriodic)
+		require.NoError(t, err)
+		err = am2ZoneA.loadAndSyncConfigs(context.Background(), reasonPeriodic)
+		require.NoError(t, err)
+		err = am1ZoneB.loadAndSyncConfigs(context.Background(), reasonPeriodic)
+		require.NoError(t, err)
+
+		require.Eventually(t, func() bool {
+			_, err1 := alertStore.GetFullState(context.Background(), user1)
+			_, err2 := alertStore.GetFullState(context.Background(), user2)
+			_, err3 := alertStore.GetFullState(context.Background(), user3)
+			return err1 == nil && err2 == nil && err3 == nil
+		}, 5*time.Second, 100*time.Millisecond, "timed out waiting for state to be persisted")
+	}
+
+	zoneATotal := 0
+	allUserIDs, configs, err := am1ZoneA.loadAlertmanagerConfigs(ctx)
+	require.NoError(t, err)
+	assert.Equal(t, len(allUserIDs), 3)
```

[nit] The arguments to `assert.Equal()` are "testing, expected, actual". So we should flip it, because the hardcoded value `3` is the expectation. In case of error, the assertion error message will be clearer to read.

```suggestion
	assert.Equal(t, 3, len(allUserIDs))
```

_Same comment applies below._